### ISSUE TITLE
Unable to ignore errors 

### ISSUE THREAD

**Issue by pociej** [rhysd/actionlint#481]
Hi I observed strange experience Im unable to reproduce. 

here is me config 

```
self-hosted-runner:
  labels:
    - aws-linux-small
    - aws-linux-medium
paths:
  .github/workflows/**/*.yaml:
    ignore:
      - 'shellcheck reported issue in this script: SC2086:.+'
      - 'shellcheck reported issue in this script: SC2046:.+'
      - 'shellcheck reported issue in this script: SC2155:.+'
      - 'shellcheck reported issue in this script: SC2044:.+'

``` 
It was working for me perfectly fine and stopped because of reasons i dont understand. 
Im not hitted by errors listed on `ignore` list 

like that 

```
.github/workflows/release.yaml:429:7: shellcheck reported issue in this script: SC2086:info:2:29: Double quote to prevent globbing and word splitting [shellcheck]
    |
429 |       run: |
    |       ^~~~
.github/workflows/test_e2e_devnet.yaml:57:9: shellcheck reported issue in this script: SC2044:warning:1:17: For loops over find output are fragile. Use find -exec or a while read loop [shellcheck]
   |
[57](https://github.com/vlayer-xyz/vlayer/actions/runs/11928877667/job/33246589658#step:4:58) |         run: |
   |         ^~~~
```

Im sure the config file is properly loaded as `self-hosted-runners` are there. 
Im running actionlint as documented : 

```
jobs:
  actionlint:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - name: Download actionlint
        id: get_actionlint
        run: bash <(curl https://raw.githubusercontent.com/rhysd/actionlint/main/scripts/download-actionlint.bash) 1.7.3
        shell: bash
      - name: Check workflow files
        run: ${{ steps.get_actionlint.outputs.executable }}
        shell: bash

```

any idea of what could be the reason? Thx in advance.

**Comment by rhysd** [comment]
Can you provide your workflow input (smaller is better) which can reproduce this issue?

**Comment by ahmedelgabri** [comment]
@pociej make sure you are targeting the correct files, I had a problem yesterday with this too, then I realized I had to change the glob to `*.{yaml,yml}`

**Comment by rhysd** [comment]
I tried to reproduce it but I could not.

`.github/actionlint.yaml`:

```yaml
paths:
  .github/workflows/**/*.yaml:
    ignore:
      - 'shellcheck reported issue in this script: SC2086:.+'
```

`.github/workflows/test.yaml`:

```yaml
on: push

jobs:
  tests:
    runs-on: ubuntu-latest
    steps:
      # This causes SC2086
      - run: echo $FOO
```

And I ran:

```sh
actionlint
```

The command did not report any error. When I removed `actionlint.yaml`, it reported an error. So the command seems to work as intended.

Can you check `actionlint` command is the latest by `actionlint -version`?

**Comment by chrisponton** [comment]
This is happening to me too.


.github/actionlint.yaml:
```YAML
paths:
  .github/workflows/**/*.yaml:
    ignore:
      - 'the runner of ".+" action is too old to run on GitHub Actions'
```

.github/workflows/simple.yml:
```YAML
on:
  pull_request:
jobs:
  checkout:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
```

Run:
```ssh
actionlint
```

Output:
```ssh
simple.yml:7:15: the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue [action]
   |
7 |       - uses: actions/checkout@v3
   |               ^~~~~~~~~~~~~~~~~~~
```

Version:
```sh
actionlint -version
1.7.4
installed by building from source
built with go1.23.2 compiler for darwin/arm64
```

**Comment by rhysd** [comment]
Thank you @chrisponton for the steps. However, I still cannot reproduce this issue... `actionlint` command reports nothing. And when removing the `- 'the runner of ".+" action is too old to run on GitHub Actions'` line, it reports one error.

Here are the steps I tried:

```sh
rm -rf ./tmp  # Ensure the directory doesn't exist
mkdir ./tmp && cd ./tmp
mkdir -p .github/workflows
vim .github/actionlint.yml  # Write the config file
vim .github/workflows/simple.yml  # Write the workflow file
git init
actionlint
```

And `actionlint` in my local:

```console
$ which actionlint
/usr/local/bin/actionlint
$ actionlint -version
1.7.4
installed by building from source
built with go1.23.2 compiler for darwin/amd64
```

Is anything different from what you tried?

**Comment by rhysd** [comment]
@chrisponton I appreciate your effort for the repro. That sounds weird. I guess there would be some hidden pre-condition. Let's keep this issue open with `not-reproducible` label so that someone else will be able to provide another repro in the future.

**Comment by pseudo-su** [comment]
I have a minimal reproduction for this here https://github.com/pseudo-su/actionlint-repro-issue-481

looks like it was fixed in versions 1.7.4 onwards

```
# Bug is present in
go run github.com/rhysd/actionlint/cmd/actionlint@v1.7.0
go run github.com/rhysd/actionlint/cmd/actionlint@v1.7.1
go run github.com/rhysd/actionlint/cmd/actionlint@v1.7.2
go run github.com/rhysd/actionlint/cmd/actionlint@v1.7.3

# Fixed in
go run github.com/rhysd/actionlint/cmd/actionlint@v1.7.4
go run github.com/rhysd/actionlint/cmd/actionlint@v1.7.5
go run github.com/rhysd/actionlint/cmd/actionlint@v1.7.6
go run github.com/rhysd/actionlint/cmd/actionlint@v1.7.7
```

**Comment by Riyan12xx** [comment]
Always on

**Comment by roryabraham** [comment]
I'm also experiencing an issue where:

- I have an actionlint.yml I know is getting picked up because the self-hosted runners I've declared there are not being reported as invaid
- running `actionlint -color -ignore '"env" is not allowed in "runs" section because .* is a Composite action.*'` passes w/ exit code 0, but not when the ignore is put in my config file:

```yml
# See https://github.com/rhysd/actionlint/blob/main/docs/config.md
self-hosted-runner:
  labels:
    - ubuntu-latest-xl
    - macos-15-large
    - macos-15-xlarge
    - ubuntu-latest-reassure-tests
    - macos-12

paths:
  .github/workflows/**/*.{yml,yaml}:
    ignore:
      - '"env" is not allowed in "runs" section because .* is a Composite action.*'
```
